src/odemis/model/_futures.py

```diff
@@ -570,6 +573,11 @@ def _on_future_done(self, f):
             # we don't require the futures to be carried out sequentially
             self.set_result(None)
 
+    def _on_batch_future_done(self, _):
+        """Called whenever the ProgressiveBatchFuture is finished."""
+        logging.debug("ProgressiveBatchFuture done.")
+        self._is_cancel_triggered = False
```

Comment:
Is that helpful? A Future can be run only once. Once it's finished, it cannot be cancelled anymore. So if it's done, there is no need to be able to run/cancel it again.
Or can you add a comment to explain why you do that?

Reply:
Right, this is not needed